clang/test/dpct/syncthreads.cu

```diff
@@ -149,4 +151,215 @@ __global__ void test9(S4 a) {
   S4 b(a);
   // CHECK:item_ct1.barrier();
   __syncthreads();
-}
\ No newline at end of file
+}
+
+extern __shared__ float extern_local_decl[];
+
+__global__ void test10(float *arg1_ptr, int arg2_scalar, int arg3_scalar) {
+  int var1 = blockIdx.x * blockDim.x + threadIdx.x;
+  if (var1 < arg3_scalar) {
+    float *var2 = extern_local_decl;
+    var2[var1] = 0;
+    if (var1 < 123) {
+      float a = __expf(1.f);
+      var2[var1] = a;
+      arg1_ptr[var1] = a;
+    }
+    // CHECK: item_ct1.barrier(sycl::access::fence_space::local_space);
```

Comment:
how about emit a DPCT diagnostic message to tell if the test10 kernel is called by 3D dim/block, it should use barrier(global_space).

Reply:
done